src/service/url-replacements-impl.js

```diff
@@ -103,8 +115,10 @@ export class UrlReplacements {
     }));
 
     // Returns the referrer URL.
-    this.set_('DOCUMENT_REFERRER', () => {
-      return viewerFor(this.win_).getReferrerUrl();
+    this.setAsync_('DOCUMENT_REFERRER', () => {
+      return viewerFor(this.win_).getReferrerUrl().then(referrer => {
+        return referrer === undefined ? '' : referrer;
```

Comment:
Could you please clarify, what's the purpose of this? Did we disallow `undefined` values downstream?